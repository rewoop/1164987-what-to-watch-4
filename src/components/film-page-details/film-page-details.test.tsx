import * as React from "react";
import * as renderer from "react-test-renderer";
import FilmPageDetails from "./film-page-details";

const Settings = {
  FILM_GENRE: `Thriller`,
  RELEASE_DATE: 1996,
  RUN_TIME: `1h 39m`,
  FILM_DIRECTOR: `Wes Andreson`,
  FILM_STARRING: [
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Tom Cruz`
  ],
};

it(`Should FilmPageDetails render correctly`, () => {
  const tree = renderer
    .create(<FilmPageDetails
      genre={Settings.FILM_GENRE}
      releaseDate={Settings.RELEASE_DATE}
      filmDirector={Settings.FILM_DIRECTOR}
      filmStarring={Settings.FILM_STARRING}
      runTime={Settings.RUN_TIME}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
