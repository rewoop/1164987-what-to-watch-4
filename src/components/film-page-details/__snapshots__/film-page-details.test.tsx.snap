// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmPageDetails render correctly 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        Wes Andreson
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
      >
        Bill Murray, Edward Norton, Jude Law, Willem Dafoe, Tom Cruz
      </span>
    </p>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
      >
        1h 39m
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        Thriller
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        1996
      </span>
    </p>
  </div>
</div>
`;
